src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs

```diff
@@ -43,6 +44,13 @@ public UISearchResourceV2(IPackageRepository repo)
         {
             return await Task.Run(() =>
             {
+                // Check if source is available.
```

Comment:
the source might be valid, but not available (network issues etc), .. if you think valid is better ill change that.